### ISSUE TITLE
edit_assignment: Resets the filter upon execution

### ISSUE THREAD

**Issue by joeltio** [joeltio/pe#11]
After editing an assignment, the `find` filter is reset.

Steps to reproduce (starting from a fresh install):
1. Create an assignment: `add_assignment si/1 an/Test Assignment ms/100`
2. Filter the first person: `find Alex`
3. Edit the assignment: `edit_assignment si/1 ai/1 an/New Assignment ms/100`
4. Note that the `find` filter is reset

After step 2:
![Screenshot 2024-11-15 at 5.08.51 PM.png](https://raw.githubusercontent.com/joeltio/pe/main/files/05a4507b-350d-46a3-9988-a7c99064c9cf.png)

After step 3 (Actual):
![Screenshot 2024-11-15 at 5.15.23 PM.png](https://raw.githubusercontent.com/joeltio/pe/main/files/06e06694-102f-4f1a-9c37-3fe856d64b4f.png)

After step 3 (Expected):
Note that the command output should be the same as the screenshot above. Other than that, the screenshot below is accurate.
![Screenshot 2024-11-15 at 5.08.51 PM.png](https://raw.githubusercontent.com/joeltio/pe/main/files/05a4507b-350d-46a3-9988-a7c99064c9cf.png)

**Comment by nus-se-script** [comment]
[IMPORTANT!: Please do not edit or reply to this comment using the GitHub UI. You can respond to it using CATcher during the next phase of the PE]

# Team's Response
This is essentially a duplicate, where the find filter is intentionally reset after each command. 


It was intentional behaviour as we believe that users would want to see the main student list after executing each command.

## The 'Original' Bug

[The team marked this bug as a duplicate of the following bug]

># grade: Resets the filter upon execution
>
>After grading an assignment, the `find` filter is reset.
>
>Steps to reproduce (starting from a fresh install):
>1. Create an assignment: `add_assignment si/1 an/Test Assignment ms/100`
>2. Filter the first person: `find Alex`
>3. Delete the assignment: `grade si/1 ai/1 s/100`
>4. Note that the `find` filter is reset
>
>After step 2:
>![Screenshot 2024-11-15 at 5.08.51 PM.png](https://raw.githubusercontent.com/nus-cs2103-AY2425S1/pe/master/files/05a4507b-350d-46a3-9988-a7c99064c9cf.png)
>
>After step 3 (Actual):
>![Screenshot 2024-11-15 at 5.23.03 PM.png](https://raw.githubusercontent.com/nus-cs2103-AY2425S1/pe/master/files/2b58e69f-fb2e-4a95-9dd4-9647ec6613f5.png)
>
>After step 3 (Expected):
>Note that the command output should be the same as the screenshot above. Other than that, the screenshot below is accurate.
>![Screenshot 2024-11-15 at 5.08.51 PM.png](https://raw.githubusercontent.com/nus-cs2103-AY2425S1/pe/master/files/05a4507b-350d-46a3-9988-a7c99064c9cf.png)
>
>
>
>
>
><hr>
><sub>[original: nus-cs2103-AY2425S1/pe-interim#700] [original labels: severity.Medium type.FunctionalityBug]</sub>
>
>
>

## Their Response to the 'Original' Bug

[This is the team's response to the above 'original' bug]

>This was intentional behaviour. 
>
>After the teacher does a "find" command, we believe the teacher would have already located the specific student they are looking for, and grade them. 
>After executing the next command, it would be more appropriate for the application to revert to listing all the students in the app. 
>
>In addition, our UG did not specify that the find filter will persist after other commands are performed. 
>
>However, future iterations can improve on this, and provide users the optionality to remain in the find filter. 
>
>
>Severity was revised down to low, since this is a minor inconvenience that occurs in rare situations.

# Items for the Tester to Verify

## :question: Issue duplicate status

Team chose to mark this issue as a duplicate of another issue (as explained in the _**Team's response**_ above)

- [ ] I disagree

**Reason for disagreement:** [replace this with your reason]

<catcher-end-of-segment><hr>

## :question: Issue response

Team chose [`response.NotInScope`]

- [ ] I disagree

**Reason for disagreement:** [replace this with your reason]

<catcher-end-of-segment><hr>

## :question: Issue severity

Team chose [`severity.Low`]
Originally [`severity.Medium`]

- [ ] I disagree

**Reason for disagreement:** [replace this with your reason]

<catcher-end-of-segment><hr>
